crates/rolldown/src/bundle/bundle_factory.rs

```diff
@@ -90,39 +91,34 @@ impl BundleFactory {
     ))
   }
 
-  pub fn create_bundle(&mut self) -> Bundle {
+  pub fn create_bundle(
```

Comment:
I removed the second `create_incremental_bundle` method, and merge the logic into one method, becuase  I found the previous pattern is bug-prone and hard to track the control flow.

It also causes a bug.

```rust
let bundler = Bundler::new(..) // with `incremental: true`

bundler.generate()
bundler.incremental_generate(["main.js"])
```

In the above code, `incremental_generate` will panic because of `generate` will call `create_bundle` instead of `create_incremental_bundle`, so it doesn't create build cache.